--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,7 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 option(UPDATE_TRANSLATIONS "Update source translation translations/*.ts files" OFF)
 
 set(LXQTBT_MINIMUM_VERSION "0.13.0")
-set(KF5_MINIMUM_VERSION "5.101.0")
+set(KF5_MINIMUM_VERSION "5.36.0")
 set(LXQT_MINIMUM_VERSION "1.4.0")
 set(QT_MINIMUM_VERSION "5.15.0")
 
--- a/src/idlenesswatcher.cpp
+++ b/src/idlenesswatcher.cpp
@@ -30,7 +30,6 @@
 #include <Solid/Device>
 #include <Solid/Battery>
 #include <KWindowSystem/KWindowSystem>
-#include <KWindowSystem/KX11Extras>
 #include <KWindowSystem/KWindowInfo>
 #include <QDebug>
 #include <LXQt/lxqtnotification.h>
@@ -162,7 +161,7 @@ void IdlenessWatcher::timeoutReached(int
 
     // check if disable Idleness when fullscreen is enabled
     if (mPSettings.isDisableIdlenessWhenFullscreenEnabled()) {
-        WId w = KX11Extras::activeWindow();
+        WId w = KWindowSystem::activeWindow();
         KWindowInfo info(w, NET::WMState);
         if (info.hasState(NET::FullScreen)) {
             QTimer::singleShot(0, this, [] {
